### ISSUE TITLE
Use pytorch early stopping and reduce LR implementation 

### ISSUE THREAD

**Issue by jeff-regier** [YosefLab/scvi-tools#393]
It'd be better to use Pytorch's `ReduceLROnPlateau` function rather than our own implementation of it. `ReduceLROnPlateau` lowers the learning rate multiple times rather than just once, so it probably improves on what we do now. 

As part of this issue, let's delete all this code:
https://github.com/YosefLab/scVI/blob/1f399e75068bd80aeb785a1810b30cdd34bfd14b/scvi/inference/trainer.py#L259-L352

**Comment by jeff-regier** [comment]
Also, we may not need to grid search over learning rates using hyperopt any more once we start using `ReduceLROnPlateau`. We can just start with a high learning rate, e.g. `1e-2`, and run until the learning rate reaches a small value, e.g., `1e-6`.

**Comment by adamgayoso** [comment]
This does seem to reduce the LR multiple times based on the patience. This implementation works and appears to mimic what's implemented in pytorch to some extent. That said, it's probably best to use their implementation.

**Comment by jeff-regier** [comment]
I see 3 major benefits from using the Pytorch implementation:

- We can start with a really high learning rate, e.g. 1e-2 and make very fast progress at first
- We can end with a really low learning rate,.e.g. 1e-6, giving us a better minima
- We can avoid grid searching over multiple learning rates, so hyperopt will run like 300% faster
